workers/unity/Assets/Gdk/Core/Tests/Editmode/GameObjectRepresentation/EntityGameObjectLifecycleSystemTests.cs

```diff
@@ -0,0 +1,62 @@
+using Improbable.Gdk.TestUtils;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class EntityGameObjectLifecycleSystemTests : HybridGdkSystemTestBase
+    {
+        private World world;
+        private EntityManager entityManager;
+        private MutableView mutableView;
+        private EntityGameObjectLifecycleSystem entityGameObjectLifecycleSystem;
+        private Entity testEntity;
+        private GameObject testGameObject;
+
+        [SetUp]
+        public void Setup()
+        {
+            world = new World("TestWorld");
+            entityManager = world.GetOrCreateManager<EntityManager>();
+            entityGameObjectLifecycleSystem = world.GetOrCreateManager<EntityGameObjectLifecycleSystem>();
+            testEntity = entityManager.CreateEntity();
+            testGameObject = new GameObject();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Object.DestroyImmediate(testGameObject);
```

Comment:
Should this check if the `testGameObject` is null?

Reply:
wrapped the destroy call in an if statement.